Makefile

```diff
@@ -338,11 +339,11 @@ endif
 # PPA includes the required mongodb-server binaries.
 install-snap-dependencies:
 ## install-snap-dependencies: Install the supported snap dependencies
-ifeq ($(shell go version | grep -o "go1.18" || true),go1.18)
-	@echo Using installed go-1.18
+ifeq ($(shell go version | cut -d ' ' -f 3 | sed -e /go/s///),$(GO_MOD_VERSION))
+	@echo 'Using installed go-$(GO_MOD_VERSION)'
 else
-	@echo Installing go-1.18 snap
-	@sudo snap install go --channel=1.18/stable --classic
+	@echo 'Installing go-$(GO_MOD_VERSION) snap'
+	@sudo snap install go --channel=$(GO_MOD_VERSION)/stable --classic
```

Comment:
This will fail if the user has a different version of go installed. I think what we should do instead is:
- If go is not installed, install from $(GO_MOD_VERSION)/stable
- If go is installed, check the version
  - If go version >= GO_MOD_VERSION, we're good
  - If go version < GO_MOD_VERSION, warn the user that they need to update to at least GO_MOD_VERSION. We shouldn't mess with a user's existing Go installation